# HG changeset patch
# User Rik <rik@octave.org>
# Date 1442925905 25200
#      Tue Sep 22 05:45:05 2015 -0700
# Node ID afdb856e44f19c16322d6a81d6ef610c1b8fc787
# Parent  1f330d33388fd65c5380a35c6632eab48103115d
Deprecate mahalanobis function.

* NEWS: Announce deprecation.

* stats.txi: Remove from manual

* scripts/deprecated/module.mk: Add to deprecated build dir.

* scripts/statistics/base/module.mk: Remove from existing dir

* scripts/deprecated/mahalanobis.m: Add warning message to deprecated function.

* scripts/statistics/base/mahalanobis.m: Delete deprecated function.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -42,16 +42,17 @@ Summary of important user-visible change
 
     The following functions have been deprecated in Octave 4.2 and will
     be removed from Octave 4.6 (or whatever version is the second major
     release after 4.2):
 
       Function             | Replacement
       ---------------------|------------------
       bitmax               | flintmax
+      mahalanobis          | mahal in Octave-Forge statistics pkg
       wavread              | audioread
       wavwrite             | audiowrite
 
  ** The following functions were deprecated in Octave 3.8 and have been
     removed from Octave 4.2.
 
       default_save_options    java_new
       gen_doc_cache           java_unsigned_conversion
diff --git a/doc/interpreter/stats.txi b/doc/interpreter/stats.txi
--- a/doc/interpreter/stats.txi
+++ b/doc/interpreter/stats.txi
@@ -132,18 +132,16 @@ different measures from those of the bas
 @DOCSTRING(runlength)
 
 @DOCSTRING(probit)
 
 @DOCSTRING(logit)
 
 @DOCSTRING(cloglog)
 
-@DOCSTRING(mahalanobis)
-
 @DOCSTRING(table)
 
 @node Statistical Plots
 @section Statistical Plots
 
 @c Should hist be moved to here, or perhaps the qqplot and ppplot
 @c functions should be moved to the Plotting Chapter?
 
diff --git a/scripts/statistics/base/mahalanobis.m b/scripts/deprecated/mahalanobis.m
rename from scripts/statistics/base/mahalanobis.m
rename to scripts/deprecated/mahalanobis.m
--- a/scripts/statistics/base/mahalanobis.m
+++ b/scripts/deprecated/mahalanobis.m
@@ -13,29 +13,41 @@
 ## General Public License for more details.
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn {Function File} {} mahalanobis (@var{x}, @var{y})
+##
+## @code{mahalanobis} is deprecated and will be removed in Octave version 4.6.
+## See the @code{mahal} function in the statistics package from Octave-Forge
+## for equivalent functionality.
+##
 ## Return the Mahalanobis' D-square distance between the multivariate
 ## samples @var{x} and @var{y}.
 ##
 ## The data @var{x} and @var{y} must have the same number of components
 ## (columns), but may have a different number of observations (rows).
 ## @end deftypefn
 
 ## Author: Friedrich Leisch <leisch@ci.tuwien.ac.at>
 ## Created: July 1993
 ## Adapted-By: jwe
 
 function retval = mahalanobis (x, y)
 
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "mahalanobis is obsolete and will be removed from a future version of Octave, please use mahal from the statistics package in Octave-Forge instead");
+  endif
+
   if (nargin != 2)
     print_usage ();
   endif
 
   if (   ! (isnumeric (x) || islogical (x))
       || ! (isnumeric (y) || islogical (y)))
     error ("mahalanobis: X and Y must be numeric matrices or vectors");
   endif
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -8,16 +8,17 @@ scripts_deprecated_FCN_FILES = \
   scripts/deprecated/find_dir_in_path.m \
   scripts/deprecated/finite.m \
   scripts/deprecated/fmod.m \
   scripts/deprecated/fnmatch.m \
   scripts/deprecated/gmap40.m \
   scripts/deprecated/isstr.m \
   scripts/deprecated/loadaudio.m \
   scripts/deprecated/luinc.m \
+  scripts/deprecated/mahalanobis.m \
   scripts/deprecated/mouse_wheel_zoom.m \
   scripts/deprecated/nfields.m \
   scripts/deprecated/octave_tmp_file_name.m \
   scripts/deprecated/playaudio.m \
   scripts/deprecated/saveaudio.m \
   scripts/deprecated/setaudio.m \
   scripts/deprecated/syl.m \
   scripts/deprecated/usage.m \
diff --git a/scripts/statistics/base/module.mk b/scripts/statistics/base/module.mk
--- a/scripts/statistics/base/module.mk
+++ b/scripts/statistics/base/module.mk
@@ -7,17 +7,16 @@ scripts_statistics_base_FCN_FILES = \
   scripts/statistics/base/cov.m \
   scripts/statistics/base/gls.m \
   scripts/statistics/base/histc.m \
   scripts/statistics/base/iqr.m \
   scripts/statistics/base/kendall.m \
   scripts/statistics/base/kurtosis.m \
   scripts/statistics/base/logit.m \
   scripts/statistics/base/lscov.m \
-  scripts/statistics/base/mahalanobis.m \
   scripts/statistics/base/mean.m \
   scripts/statistics/base/meansq.m \
   scripts/statistics/base/median.m \
   scripts/statistics/base/mode.m \
   scripts/statistics/base/moment.m \
   scripts/statistics/base/ols.m \
   scripts/statistics/base/ppplot.m \
   scripts/statistics/base/prctile.m \
